lib/Search/AOD/LuceneSearchEngine.php

```diff
@@ -0,0 +1,178 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\Search\AOD;
+
+use ACLController;
+use BeanFactory;
+use SecurityGroup;
+use stdClass;
+use SugarBean;
+use SuiteCRM\Exception\Exception;
+use SuiteCRM\Search\SearchEngine;
+use SuiteCRM\Search\SearchQuery;
+use SuiteCRM\Search\SearchResults;
+
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+/**
+ * Class LuceneSearchEngine
+ * @package SuiteCRM\Search\AOD
+ */
+class LuceneSearchEngine extends SearchEngine
+{
+    /** @var bool|SugarBean */
+    private $index;
+
+    public function __construct()
+    {
+        $this->index = BeanFactory::getBean("AOD_Index")->getIndex();
+    }
+
+    /**
+     * Search function run when user goes to Show All and runs a search again.  This outputs the search results
+     * calling upon the various listview display functions for each module searched on.
+     *
+     * @param SearchQuery $query
+     *
+     * @return SearchResults
+     * @throws Exception
+     */
+    public function search(SearchQuery $query): SearchResults
+    {
+        $queryString = $query->getSearchString();
+
+        $start = microtime(true);
+        $hits = $this->runLucene($queryString);
+        $results = $this->parseHits($hits);
+        $end = microtime(true);
+        $elapsed = $end - $start;
+
+        return new SearchResults($results['modules'], true, $elapsed, count($results['hits']));
+    }
+
+    /**
+     *
+     * @param string $queryString
+     * @return array
+     */
+    private function runLucene(string $queryString): array
+    {
+        $cachePath = 'cache/modules/AOD_Index/QueryCache/' . md5($queryString);
+        if (is_file($cachePath)) {
+            $mTime = filemtime($cachePath);
+            if ($mTime > (time() - 5 * 60)) {
```

Comment:
Hi @Dillon-Brown, does this mean we are caching query results for a given amount of time?

Reply:
Yes, I've kept this functionality the same as before, only real change here is that I'm calling ```filemtime``` directly instead of using a method to calculate DST manually (Due to a PHP bug a decade ago).

Reply:
Hi @Dillon-Brown, thank you. I maybe wrong, though Ideally we should cache the query results? we may get results that are out of date right?

Reply:
@Dillon-Brown as you mentioned this is keeping existing behavior. This is something we can go over on a separate conversation